### ISSUE TITLE
Clipping an Image with a Border has a gap on Windows

### ISSUE THREAD

**Issue by tchtang** [dotnet/maui#9129]
### Description

XAML is

```
    <Grid BackgroundColor="Blue">
        <Grid.RowDefinitions>
            <!-- label, search -->
            <RowDefinition Height="Auto" />
        </Grid.RowDefinitions>
        <HorizontalStackLayout Grid.Row="0" Margin="20">
            <!-- meal all -->
            <Border Stroke="LightGray" StrokeThickness="3">
                <Border.StrokeShape>
                    <RoundRectangle CornerRadius="20" />
                </Border.StrokeShape>
                <VerticalStackLayout BackgroundColor="#f0f0f0">
                    <Image Aspect="AspectFill" HeightRequest="100" Source="meal_all.jpg" WidthRequest="100" />
                    <Label HorizontalTextAlignment="Center" Margin="0,5" Text="All" />
                </VerticalStackLayout>
            </Border>
            <!-- meal noodle -->
            <Border Stroke="LightGray" StrokeThickness="3">
                <Border.StrokeShape>
                    <RoundRectangle CornerRadius="20" />
                </Border.StrokeShape>
                <Grid BackgroundColor="#f0f0f0">
                    <Grid.RowDefinitions>
                        <RowDefinition Height="Auto" />
                        <RowDefinition Height="Auto" />
                    </Grid.RowDefinitions>
                    <Image Grid.Row="0" Aspect="AspectFill" HeightRequest="100" Source="meal_noodle.jpg" WidthRequest="100" />
                    <Label Grid.Row="1" HorizontalTextAlignment="Center" Margin="0,5" Text="Noodle" />
                </Grid>
            </Border>
            <!-- meal rice -->
            <Border Stroke="LightGray" StrokeThickness="3">
                <Border.StrokeShape>
                    <RoundRectangle CornerRadius="20" />
                </Border.StrokeShape>
                <VerticalStackLayout BackgroundColor="#f0f0f0">
                    <Image Aspect="AspectFill" HeightRequest="100" Source="meal_rice.jpg" WidthRequest="100" />
                    <Label HorizontalTextAlignment="Center" Margin="0,5" Text="Rice" />
                </VerticalStackLayout>
            </Border>
        </HorizontalStackLayout>
    </Grid>
```

Android API 30 is okay

![image](https://user-images.githubusercontent.com/109950893/182313877-023afad1-c884-4f21-b348-a050268d6ce9.png)

Window 10 has gap

![image](https://user-images.githubusercontent.com/109950893/182314035-abd9a3db-a1bb-4478-ad76-44a2a5db1819.png)

### Public API Changes

.

### Intended Use-Case

please reduce gap within Border

**Comment by breenbob** [comment]
This is affecting me as well. I've used Border with RadioButton and VisualStateManager to create a custom segmented control, but I'm seeing different behaviour with Border on every platform:

Windows (dark mode but clearly same issue as above):
![image](https://user-images.githubusercontent.com/6000851/212373991-9fc7c683-9ff0-4534-bae8-b8e08ba8f96a.png)

Android (Spot on):
![image](https://user-images.githubusercontent.com/6000851/212374550-285bf6dc-df10-4d62-ba24-6def71881fec.png)

iOS (slightly different issue, acts as if padding set but all controls have Padding=0, and only TL and TR corners are clipped):
![image](https://user-images.githubusercontent.com/6000851/212387885-afa11c42-af56-470c-9658-4b11f8082496.png)

**Comment by breenbob** [comment]
Based on my screenshots I think possibly title should be modified to include iOS and iOS label added.

**Comment by jsuarezruiz** [comment]
Reproduced here https://github.com/jsuarezruiz/netmaui-finance-app-challenge
![image](https://user-images.githubusercontent.com/6755973/223440901-8dfb91e1-ba97-421e-ad89-3a666769a1a7.png)

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.7.0 Preview 1.0. Repro on Windows 11 and iOS 16.4 with below Project:

Windows 11: Repro
![image](https://github.com/dotnet/maui/assets/97647558/30e07e2d-75e7-43e8-b42e-c46fe88040c8)

Android 13.0: Not repro.
![image](https://github.com/dotnet/maui/assets/97647558/5d4e766b-34f1-4c6e-ac01-381a59ccb09e)

iOS: Repro
![image](https://github.com/dotnet/maui/assets/97647558/123f30c9-7cd9-4a06-bb06-fedfa52f69c2)

**Comment by breenbob** [comment]
@jsuarezruiz Is this issue also fixed with your PR https://github.com/dotnet/maui/pull/10964?

**Comment by jsuarezruiz** [comment]
> @jsuarezruiz Is this issue also fixed with your PR #10964?

No, that one applied changes mostly for iOS. Will be fixed by https://github.com/dotnet/maui/pull/14403
